srcDoc/custom_types.rst

```diff
@@ -0,0 +1,578 @@
+                success = false;
+                fprintf(stderr, "longSeq.ensure_length failed.\n");
+            }
+            DDS_ReturnCode_t retcode = data.clear_all_members();
+            if (retcode != DDS_RETCODE_OK) {
+                fprintf(stderr, "clear_all_members failed: %d.\n", retcode);
+                success = false;
+            }
+            return success;
+        }
+
+3. Build *RTI Perftest* using ``--customType <type>``.
+
+.. code-block:: bash
+
+        ./build.sh --platform x64Linux3gcc5.4.0 --nddshome /home/rti_connext_dds-6.0.0 --cpp-build --customType Test
+
+4. Launch *RTI Perftest*.
+
+.. code-block:: bash
+
+        ~/rtiperftest$ ./bin/x64Linux3gcc5.4.0/release/perftest_cpp -pub -executionTime 60 -noprint
+        RTI Perftest 3.0.0 (RTI Connext DDS 6.0.0)
+
+        Custom Type provided: 'Test'
+
+        Mode: THROUGHPUT TEST
+            (Use "-latencyTest" for Latency Mode)
+
+        Perftest Configuration:
+            Reliability: Reliable
+            Keyed: No
+            Publisher ID: 0
+            Latency count: 1 latency sample every 10000 samples
+            Data Size: 100
+            Batching: 8192 Bytes (Use "-batchSize 0" to disable batching)
+            Publication Rate: Unlimited (Not set)
+            Execution time: 60 seconds
+            Receive using: Listeners
+            Domain: 1
+            Dynamic Data: No
+            Asynchronous Publishing: No
+            XML File: perftest_qos_profiles.xml
+
+        Transport Configuration:
+            Kind: UDPv4 & SHMEM (taken from QoS XML file)
+            Use Multicast: False
+
+        Waiting to discover 1 subscribers ...
+        Waiting for subscribers announcement ...
+        Sending 4050 initialization pings ...
+        Publishing data ...
+        Setting timeout to 60 seconds
+        Length:   464  Latency: Ave     39 us  Std   30.7 us  Min     21 us  Max    276 us  50%     30 us  90%     60 us  99%    276 us  99.99%    276 us  99.9999%    276 us
+        Finishing test due to timer...
+        Test ended.
+
+
+.. code-block:: bash
+
+        ~/rtiperftest$ ./bin/x64Linux3gcc5.4.0/release/perftest_cpp -sub -noprint
+        RTI Perftest 3.0.0 (RTI Connext DDS 6.0.0)
+
+        Custom Type provided: 'Test'
+
+        Perftest Configuration:
+            Reliability: Reliable
+            Keyed: No
+            Subscriber ID: 0
+            Receive using: Listeners
+            Domain: 1
+            Dynamic Data: No
+            XML File: perftest_qos_profiles.xml
+
+        Transport Configuration:
+            Kind: UDPv4 & SHMEM (taken from QoS XML file)
+            Use Multicast: False
+
+        Waiting to discover 1 publishers ...
+        Waiting for data ...
+        Length:   464  Packets: 68081040  Packets/s(ave): 1134692  Mbps(ave):  4212.0  Lost:     0 (0.00%)
+        Finishing test...
+        Test ended.
+
+
+
+5. You can also launch *RTI Perftest* with your customType using DynamicData.
+
```

Comment:
In this mode, Perftest will use the DynamicData API to serialize/deserialize the customType data structure instead of the generated serialization/deserialization code.